src/test/java/com/amazon/opendistroforelasticsearch/sql/unittest/ExplainTest.java

```diff
@@ -0,0 +1,115 @@
+import static org.hamcrest.Matchers.equalTo;
+
+public class ExplainTest {
+
+    @Test
+    public void assertExplainRequestShouldBeJson() throws IOException {
+
+        String expectedOutputFilePath = TestUtils.getResourceFilePath(
+                "src/test/resources/expectedOutput/explain_format_pretty.json");
+        String expectedOutput = Files.toString(new File(expectedOutputFilePath), StandardCharsets.UTF_8);
+
+        String explainResult = explain(String.format("{\"query\":\"" +
+                "SELECT firstname " +
+                "FROM %s\"}", TestsConstants.TEST_INDEX_ACCOUNT));
+        String result = new JsonPrettyFormatter().format(explainResult);
+
+        assertThat(result.replaceAll("\\s", ""),
+                equalTo(expectedOutput.replaceAll("\\s", "")));
+    }
+
+    @Test
+    public void assertExplainRequestPrettyFormatted() throws IOException {
+
+        String expectedOutputFilePath = TestUtils.getResourceFilePath(
+                "src/test/resources/expectedOutput/explain_format_pretty.json");
+        String expectedOutput = Files.toString(new File(expectedOutputFilePath), StandardCharsets.UTF_8);
+
+        String explainResult = explain(String.format("{\"query\":\"" +
+                "SELECT firstname " +
+                "FROM %s\"}", TestsConstants.TEST_INDEX_ACCOUNT));
+        String result = new JsonPrettyFormatter().format(explainResult);
+
+        assertThat(result, equalTo(expectedOutput));
+    }
+
+    @Test
+    public void assertExplainJoinRequestPrettyFormatted() throws IOException {
+
+        String expectedOutPutFilePath = TestUtils.getResourceFilePath(
+                "src/test/resources/expectedOutput/explain_join_format_pretty.json");
+        String expectedOutput = Files.toString(new File(expectedOutPutFilePath), StandardCharsets.UTF_8);
+
+        String bank = TestsConstants.TEST_INDEX_BANK;
+        String explainResult = explain(String.format("{\"query\":\"" +
+                "SELECT b1.firstname, b1.lastname, b2.age  " +
+                "FROM %s b1 " +
+                "LEFT JOIN %s b2 " +
+                "ON b1.age = b2.age AND b1.state = b2.state\"}", bank, bank));
+        String result = new JsonPrettyFormatter().format(explainResult);
+
+        assertThat(result, equalTo(expectedOutput));
+    }
+
+    private String explain(String request) {
+        try {
+            JSONObject jsonRequest = new JSONObject(request);
+            String sql = jsonRequest.getString("query");
+
+            return translate(sql, jsonRequest);
+        } catch (SqlParseException | SQLFeatureNotSupportedException e) {
+            throw new ParserException("Illegal sql expr in request: " + request);
+        }
+    }
+
+    private String translate(String sql, JSONObject jsonRequest) throws SQLFeatureNotSupportedException, SqlParseException {
+        Client mockClient = Mockito.mock(Client.class);
+        CheckScriptContents.stubMockClient(mockClient);
```

Comment:
Check if this line is actually needed. In most of cases it was working without it. Also I'm sure that I saw similar code in other tests as well - consider reusing it instead of copy-pasting